main.cpp

```diff
@@ -76,16 +82,32 @@ while(window.isOpen()){
             }
         break;
         case 3:
-             if(event.type==sf::Event::KeyPressed){
-                if(event.key.code==sf::Keyboard::Escape) game.paused=!game.paused;
-           }
-            if(event.type==sf::Event::TextEntered){
-        }
-        break;
+            if (event.type == sf::Event::KeyPressed) {
+
+                // Menu to be presented upon pressing escape
+                if (event.key.code == sf::Keyboard::Escape) {
+                    
+                    if (!game.paused)
+                        escapeKeyPressed = true;
+                    else
+                        escapeKeyPressed = false;
+
+                    game.paused = !game.paused;
+
+                }
+            }
```

Comment:
It is still possible to press escape after the player's health goes down to 0.
This has to be fixed.
[Screencast from 02.06.2023 16:06:26.webm](https://github.com/sebe324/retro-game/assets/58781463/852dc10f-d69b-4295-893a-9d6c09a6f371)